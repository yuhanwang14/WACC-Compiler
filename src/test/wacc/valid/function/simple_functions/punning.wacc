# Functions should be able to have the same name as variables
# Thanks to Jordan Hall

# Output:
# 0
#

begin
    int inc(int x) is
        return x + 1
    end
    int inc = 0;
    int tmp = call inc(inc);
    println inc
end
