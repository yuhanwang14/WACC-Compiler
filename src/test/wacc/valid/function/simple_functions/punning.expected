0


begin
    int inc(int x) is
        return x + 1
    end
    int inc = 0;
    int tmp = call inc(inc);
    println inc
end